src/Features/CSharpTest/ExtractMethod/ExtractMethodTests2.cs

```diff
@@ -5453,4 +5481,389 @@ static void NewMethod()
+                    (var anotherValue, Customer customer) = await {|Rename:NewMethod|}(value);
+            
+                    Console.Write(customer.Id);
+                    Console.Write(anotherValue);
+                }
+
+                private static async Task<(string anotherValue, Customer customer)> NewMethod(string value)
+                {
+                    var anotherValue = "GooBar";
+                    var customer = await _repository.GetValue(value.Length);
+                    return (anotherValue, customer);
+                }
+            }
+            """,
+            options: ImplicitForBuiltInTypes());
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/64597")]
+    public Task TestMultipleRefCapture()
+        => TestInRegularAndScriptAsync(
+            """
+            using System;
+            using System.Threading.Tasks;
+
+            class Repository
+            {
+                public async Task Goo()
+                {
+                    int a = 0;
+                    int b = 0;
+
+                    [|a++;
+                    b++;
+                    await Goo();|]
+
+                    Console.Write(a);
+                    Console.Write(b);
+                }
+            }
+            """,
+            """
+            using System;
+            using System.Threading.Tasks;
+            
+            class Repository
+            {
+                public async Task Goo()
+                {
+                    int a = 0;
+                    int b = 0;
+                    (a, b) = await {|Rename:NewMethod|}(a, b);
+            
+                    Console.Write(a);
+                    Console.Write(b);
+                }
+
+                private async Task<(int a, int b)> NewMethod(int a, int b)
+                {
+                    a++;
+                    b++;
+                    await Goo();
+                    return (a, b);
+                }
+            }
+            """,
+            options: ImplicitTypeEverywhere());
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/64597")]
+    public Task TestMultipleRefCapture_PartialCapture()
+        => TestInRegularAndScriptAsync(
+            """
+            using System;
+            using System.Threading.Tasks;
+
+            class Repository
+            {
+                public async Task Goo()
+                {
+                    int a = 0;
+                    [|int b = 0;
+
+                    a++;
+                    b++;
+                    await Goo();|]
+
+                    Console.Write(a);
+                    Console.Write(b);
+                }
+            }
+            """,
+            """
+            using System;
+            using System.Threading.Tasks;
+            
+            class Repository
+            {
+                public async Task Goo()
+                {
+                    int a = 0;
+                    (a, var b) = await {|Rename:NewMethod|}(a);
```

Comment:
Very nice! Although, it'd be closer to the user's style to generate `(a, int b)`, since that's how `b` was declared in the original code.

Reply:
currently the tests validate we abide by the editor config options set by the user.  i am open to our systems inferring appropriate style in teh absence of any editor config settings.

*however* i woudl like to do that externally to ExtractMethod, and just make that part of how all our codegen layers work.  RIght now EM is just operating like all other features in this regard.

Reply:
> however i woudl like to do that externally to ExtractMethod, and just make that part of how all our codegen layers work

Of course! That's why I added this to my review above:

> Looks awesome! I added few thoughts and suggestions. Some might just be issues to file for later if they're interesting enough.

Reply:
Sounds good.  We can wait on feedback here.  Generally speaking, it appears as the industry/ecosystem pivoted *hard* to use editorconfig.  So the use case of needing to preserve code that is out of spec with editor config dropped majorly in priority.